catalyst/src/gemini.test.js

```diff
@@ -0,0 +1,124 @@
+import validate_questionnaire_format from './gemini';
+import validate_task_format from './gemini';
+import createPlan from './gemini';
+import Questionnaire from './Questionnaire';
+
+//TESTS THE VALIDATE_QUESTIONAIRRE FORMAT FUNCTION
+test('null input', () => {
+    const questionnaire = null;
+    boolean format = validate_questionnaire_format(questionnaire);
+    expect (format.toBe (false));
+});
+
+test('empty input', () => {
+  const questionnaire = "";
+  boolean format = validate_questionnaire_format(questionnaire);
+  expect (format.toBe (false));
+});
+
+test('array size 5 input', () => {
+  const questionnaire = { 0:['1','2','3','4','5']};
+  boolean  format = validate_questionnaire_format(questionnaire);
+  expect(format.toBe(false));
+});
+
+test('string input', () => {
+  const questionnaire = "Hello World";
+  boolean format = validate_questionnaire_format(questionnaire);
+  expect(format.toBe(false));
+});
+
+test('array size one', () => {

```

Comment:
you are using str which is not defined in that scope. It should be questionnaire.